extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,286 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Azure.Functions.Worker.Extensions;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Converter to bind Blob Storage type parameters.
+    /// </summary>
+    [SupportsDeferredBinding]
+    internal class BlobStorageConverter : IInputConverter
+    {
+        private readonly IOptions<WorkerOptions> _workerOptions;
+        private readonly IOptionsSnapshot<BlobStorageBindingOptions> _blobOptions;
+        private readonly ILogger<BlobStorageConverter> _logger;
+        private readonly Regex BlobIsFileRegex = new Regex(@"\.[^.\/]+$");
+
+        public BlobStorageConverter(IOptions<WorkerOptions> workerOptions, IOptionsSnapshot<BlobStorageBindingOptions> blobOptions, ILogger<BlobStorageConverter> logger)
+        {
+            _workerOptions = workerOptions ?? throw new ArgumentNullException(nameof(workerOptions));
+            _blobOptions = blobOptions ?? throw new ArgumentNullException(nameof(blobOptions));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            return context?.Source switch
+            {
+                ModelBindingData binding => await ConvertFromBindingDataAsync(context, binding),
+                _ => ConversionResult.Unhandled(),
+            };
+        }
+
+        private async ValueTask<ConversionResult> ConvertFromBindingDataAsync(ConverterContext context, ModelBindingData modelBindingData)
+        {
+            try
+            {
+                if (modelBindingData.Source is not Constants.BlobExtensionName)
+                {
+                    throw new InvalidBindingSourceException(Constants.BlobExtensionName);
+                }
+
+                BlobBindingData blobData = GetBindingDataContent(modelBindingData);
+                var result = await ConvertModelBindingDataAsync(context.TargetType, blobData);
+
+                if (result is null)
+                {
+                    return ConversionResult.Failed(new InvalidOperationException($"Unable to convert blob binding data to type '{context.TargetType.Name}'."));
+                }
+
+                return ConversionResult.Success(result);
+            }
+            catch (JsonException ex)
+            {
+                string msg = string.Format(CultureInfo.CurrentCulture,
+                    @"Binding parameters to complex objects uses JSON serialization.
```

Comment:
IIRC this string is used for other converters too. It might be useful to have this as a const in a shared file to standardize the message

Reply:
There are a couple constants improvements we can make so I'll leave that for a new PR